code/game/machinery/atmoalter/canister.dm

```diff
@@ -258,6 +264,22 @@ update_flag
 		..()
 		return
 
+	else if(QUALITY_PULSING in I.tool_qualities)
+		if(I.use_tool(user, src, WORKTIME_NORMAL, QUALITY_PULSING, FAILCHANCE_EASY,  required_stat = STAT_MEC))
+			if(valve_open == 1)
+				to_chat(user, SPAN_WARNING("You can't seal the gasket while the valve is open!"))
+				return
+			else if(sealed == FALSE)
+				to_chat(user, "You seal the gasket with a pulse of electricity.")
+				sealed = TRUE
+				SSnano.Destroy(src)
+				SSnano.update_uis(src) // exploit protection. no leaving the window open after sealing the can.
```

Comment:
This seems odd.
Why not check it in Topic instead? The way it works you wouldn't be able to see contents / pressure etc because gasket is sealed.